cmp/grid/GridModel.js

```diff
@@ -1031,6 +1032,93 @@ export class GridModel extends HoistModel {
         }
     }
 
+    /**
+     * Begin an inline editing session.
+     * @param {RecordOrId} [recOrId] - Record/ID to edit. If unspecified, the first selected Record
+     *      will be used, if any, or the first overall Record in the grid.
+     * @param {string} [colId] - ID of column on which to start editing. If unspecified, the first
+     *      editable column will be used.
+     * @return {Promise<void>}
+     */
+    async beginEditAsync({record, colId} = {}) {
+        const isReady = await this.whenReadyAsync();
+        if (!isReady) return;
+
+        const {store, agGridModel, agApi, selection} = this;
+
+        let recToEdit;
+        if (record) {
+            // Normalize specified record, if any.
+            recToEdit = record.isRecord ? record : store.getById(record);
+        } else {
+            if (!isEmpty(selection)) {
+                // Or use first selected record, if any.
+                recToEdit = selection[0];
+            } else {
+                // Or use the first record overall.
+                const firstRowId = agGridModel.getFirstSelectableRowNodeId();
+                recToEdit = store.getById(firstRowId);
+            }
+        }
+
+        const rowIndex = agApi.getRowNode(recToEdit?.id)?.rowIndex;
+        if (isNil(rowIndex) || rowIndex < 0) {
+            console.warn(
+                'Unable to start editing - ' +
+                record ? 'specified record not found' : 'no records found'
+            );
+            return;
+        }
+
+        let colToEdit;
+        if (colId) {
+            // Ensure specified column is editable for recToEdit.
+            const col = this.getColumn(colId);
+            colToEdit = col?.isEditableForRecord(recToEdit) ? col : null;
+        } else {
+            // Or find the first editable col in the grid.
+            colToEdit = this.getVisibleLeafColumns().find(column => {
+                return column.isEditableForRecord(recToEdit);
+            });
+        }
+
+        if (!colToEdit) {
+            console.warn(
+                'Unable to start editing - ' +
+                (colId ? `column with colId ${colId} not found, or not editable` : 'no editable columns found')
+            );
+            return;
+        }
+
+        agApi.startEditingCell({
+            rowIndex,
+            colKey: colToEdit.colId
+        });
+    }
+
+    /**
+     * Stop an inline editing session, if one is in-progress.
+     * @param {boolean} dropPendingChanges - true to cancel current edit without saving pending
+     *      changes in the active editor(s) to the backing Record.
+     * @return {Promise<void>}
+     */
+    async endEditAsync(dropPendingChanges = false) {
+        const isReady = await this.whenReadyAsync();
+        if (!isReady) return;
+
+        this.agApi.stopEditing(dropPendingChanges);
+    }
+
+    /** @package */
```

Comment:
well, this is a new jsdoc pattern

Reply:
can't these handlers be annotated with @action?

Reply:
Yes, will just need to wrap when applied in comp code to get the scope right - can do that if preferred

For clarity, `@package` is to declare "these are called outside of this class/file, but they aren't considered a part of the general API". We have only two usages in hoist already, but I think it's helpful enough to use where appropriate